test/unit/test_pipelining.cpp

```diff
@@ -540,7 +540,7 @@ bool memory_test(memtest settings) {
 	pipeline p =
 		make_pipe_begin<memtest_1, memtest &>(settings)
 		| make_pipe_end<memtest_2, memtest &>(settings);
-	p(0, pi, settings.totalMemory, TPIE_FSI);
```

Comment:
This should not be needed, we need a version of operator(), that does this automatically, otherwise we break a lot of code.